cmd/fetchbranches/main.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+func main() {
+	ci := NewCI()
+
+	if ci == nil {
+		fmt.Fprintf(os.Stderr, "CI not found\n")
+		os.Exit(1)
+	}
+
+	// Shows PR number
+	pr, err := ci.GetPR()
+
+	if err != nil {
+		fmt.Printf("%s\n", err)
+		os.Exit(1)
+	}
+	fmt.Printf("PULL REQUEST NUMBER : %d\n", pr)
+
+	// Shows owner and the repository name
+	owner, repo, err := ci.GetRepository()
+
+	if err != nil {
+		fmt.Printf("%s\n", err)
+		os.Exit(1)
+	}
+	fmt.Printf("OWNER: %s, REPOSITORY: %s\n", owner, repo)
+
+	// Shows author PR comments
+	cvr := NewCVR(CVRGithub)
+	comments, err := cvr.GetPRAuthorComments(owner, repo, pr)
+	fmt.Printf("%#v\n", comments)
+
+	// Obtains the repo and the URL where a PR will be fetched
+	fetchMap := make(map[string]string)
+
+	for _, a := range comments {
+		branches := GetBranch(a)
+		for _, b := range branches {
+			fetchMap[b.Repo] = b.URL
+		}
+	}
+
+	// Checks that GOPATH exists
+	gopath := os.Getenv("GOPATH")
+	if gopath == "" {
+		fmt.Printf("GOPATH variable is empty\n")
+		os.Exit(1)
+	}
+
+	for branchRepo, branchURL := range fetchMap {
+		// Clones the repository that is in the PR (branch name)
+		urlCVR := "github.com"
+		testRepository := filepath.Join(urlCVR, owner, branchRepo)
+		cmd := exec.Command("go", "get", "-d", testRepository)
+		err := cmd.Run()
+
+		if err != nil {
+			fmt.Printf("it was not possible to clone the repository\n")
+			os.Exit(1)
+		}
+
+		// Obtains the name of the branch i.e. topic/tests
+		// from the URL
+		wordKey := "/tree/"
+		i := strings.Index(branchURL, wordKey)
+
+		if i == -1 {
+			fmt.Printf("the word tree was not found\n")
```

Comment:
This is rather an amusing error message. It's a little too literal though :-)

I'd say something like:

```
fmt.Printf("Failed to establish branch name from URL: %v\n", branchURL)
```